From ddfd896532d37f7fa9d9a83187bbad4a9edf2984 Mon Sep 17 00:00:00 2001
From: MarvinCao <caogang440@163.com>
Date: Tue, 20 Oct 2015 12:52:57 +0800
Subject: [PATCH] =?UTF-8?q?add=5Fbss=5Fconfig=5Ffile=E5=87=BD=E6=95=B0?=
 =?UTF-8?q?=E6=9C=89bug?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

---
 src/odin/odinagent.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/odin/odinagent.c b/src/odin/odinagent.c
index 0b54d2e..62ee738 100644
--- a/src/odin/odinagent.c
+++ b/src/odin/odinagent.c
@@ -251,6 +251,12 @@ char *add_bss_config_file(const char *ctrl_config_path, OdinStationState *state,
             fputs(buf_write, fp_write);
            	continue;
         }
+        else if (strstr(buf_read_pos, "ignore_broadcast_ssid=") != NULL) {
+        	sprintf(buf_write, "%s", buf_read);
+        	//printf("%s", buf_read);
+        	fputs(buf_write, fp_write);
+        	continue;
+        }
         else if (strstr(buf_read_pos, "ssid=") != NULL) {
             strsep(&buf_read_pos, "=");
 
-- 
1.9.4.msysgit.2

